File: app/src/main/java/com/seki/saezurishiki/view/fragment/list/HomeTimeLineFragment.java
package com.seki.saezurishiki.view.fragment.list;

import android.os.Bundle;

import com.seki.saezurishiki.file.SharedPreferenceUtil;
import com.seki.saezurishiki.network.twitter.AsyncTwitterTask;

import twitter4j.Paging;
import twitter4j.ResponseList;
import twitter4j.Status;


/**
 * ホームタイムラインFragment<br>
 * ネットワーク接続時には常時ユーザーストリームで,ログインユーザーのhome time-lineを表示します
 * @author seki
 */
public class HomeTimeLineFragment extends UserStreamTimeLineFragment {


    public static TimeLineFragment getInstance() {
        return new HomeTimeLineFragment();
    }


    @Override
    void releaseSavedStatus() {
        if (mSavedStatuses.isEmpty()) {
            return;
        }

        for (long statusId : mSavedStatuses) {
            mAdapter.insert(statusId, 0);
        }

        mAdapter.notifyDataSetChanged();
        mSavedStatuses.clear();
    }


    @Override
    public void onStop() {
        if (!mAdapter.isEmpty()) {
            SharedPreferenceUtil.writeLatestID(getActivity(), SharedPreferenceUtil.HOME, mAdapter.lastReadId());
        }
        super.onStop();
    }


    @Override
    protected long readLastID() {
        return SharedPreferenceUtil.readLatestID(getActivity(), SharedPreferenceUtil.HOME);
    }


//    @Override
//    public void onStatus(final Status status) {
//        super.onStatus(status);
//        if (!mListTopVisible || mIsLoading) {
//            mSavedStatuses.add(status.getId());
//            this.tabViewControl.requestChangeTabState(this);
//            return;
//        }
//
//        mAdapter.insert(status.getId(), 0);
//        if (!this.tabViewControl.isCurrentSelect(this)) {
//            this.tabViewControl.requestChangeTabState(this);
//        }
//    }
}

File: app/src/main/java/com/seki/saezurishiki/view/fragment/list/HomeTimeLineFragment.kt
package com.seki.saezurishiki.view.fragment.list

import android.os.Bundle
import com.seki.saezurishiki.application.SaezurishikiApp
import com.seki.saezurishiki.view.HomeModule

class HomeTimeLineFragment : UserStreamTimeLineFragment() {

    companion object {
        @JvmStatic
        fun getInstance(userId: Long, tabPosition: Int, listName: String): TweetListFragment {
            return HomeTimeLineFragment().apply {
                arguments = Bundle().apply {
                    putLong(TweetListFragment.USER_ID, userId)
                    putInt(TAB_POSITION, tabPosition)
                    putString(LIST_NAME, listName)
                }
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val listOwnerId = arguments!!.getLong(TweetListFragment.USER_ID)
        SaezurishikiApp.mApplicationComponent.homeComponentBuilder()
                .listOwnerId(listOwnerId)
                .presenterView(this)
                .module(HomeModule())
                .build()
                .inject(this)
    }
}